### PR TITLE
[Rule34video] Add extractor

### PR DESCRIPTION
**Pull request by trassshhub**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Add support for downloading rule34video.com videos (Rule34VideoIE).
This is my first extractor writing, please advise if any improvement is required.

### CODE REVIEW

File: yt_dlp/extractor/rule34video.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from yt_dlp.utils import parse_duration
+
+from .common import InfoExtractor
+
+import re
```
Comment (pukkandan) [inline yt_dlp/extractor/rule34video.py:8]:
```suggestion
import re

from .utils import parse_duration
from .common import InfoExtractor
```

File: yt_dlp/extractor/rule34video.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+
+class Rule34VideoIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?rule34video\.com/videos/(?P<id>\d+)'
+    _TESTS = [
+        {
+            'url': 'https://rule34video.com/videos/3065157/shot-it-mmd-hmv/',
```
Comment (pukkandan) [inline yt_dlp/extractor/rule34video.py:15]: Add md5
```py
            'url': 'https://rule34video.com/videos/3065157/shot-it-mmd-hmv/',
            'md5': ...
```

File: yt_dlp/extractor/rule34video.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        },
+    ]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        formats = []
+
+        for mobj in re.finditer(r'<a[^>]+href="([^"]+download=true[^"]+)".*>([^\s]+) ([^<]+)p</a>', webpage):
```
Comment (pukkandan) [inline yt_dlp/extractor/rule34video.py:62]: use named groups

**Comment by trassshhub** [comment]
updated code as suggested

**Comment by afterdelight** [comment]
could you add models or members support?

**Pull request merged**